sbt_for_bazel.bzl

```diff
@@ -0,0 +1,75 @@
+def _scala_major(ver):
+  parts = ver.split(".")
+  return "%s.%s" % (parts[0], parts[1])
+
+def _clean(target_name):
+  return target_name.replace("-", "_")
+
+def _jar_path(target, version, scala):
+  return "{target}/target/scala-{scala_version}/{target}_{scala_version}-{version}.jar".format(
+      target=target,
+      scala_version=_scala_major(scala),
+      version=version)
+
+
+def _sbt_jar(ctx):
+  ctx.download_and_extract(
+      ctx.attr.sbt_url,
+      ctx.path("sbt"),
+      ctx.attr.sbt_sha256,
+      '',
+      '')
+  targets = " ".join([ "%s/package" % t for t in ctx.attr.targets ])
+
+  result = ctx.execute(
+      ["bash", "-c", """
+set -ex
+( cd {working_dir} &&
+    if ! ( cd '{dir}' && {git} rev-parse --git-dir ) >/dev/null 2>&1; then
+      rm -rf '{dir}'
+      {git} clone '{remote}' '{dir}'
+    fi
+    cd '{dir}'
+    {git} reset --hard {ref} || ({git} fetch && {git} reset --hard {ref})
+    {git} clean -xdf
+    {java} -jar {sbt_path} ++{scala} {targets})
+""".format(
+    dir=ctx.path("git"),
+    git=ctx.which("git"),
+    java=ctx.which("java"),
+    working_dir=ctx.path("git").dirname,
+    sbt_path=ctx.path("sbt/sbt/bin/sbt-launch.jar"), # we need to keep a private sbt dir
```

Comment:
We'll need to be a bit careful about how we invoke SBT, since it uses a very weird mix of global (per-machine), per-user, per-install, and per-project configuration. Like, I'm fairly sure by default it'll throw in some configurations in /etc/sbtopts or something somewhere.